argo/providers/vai/provider.go

```diff
@@ -193,8 +202,40 @@ func (vaip VAIProvider) DeletePipeline(ctx context.Context, providerConfig VAIPr
 	return nil
 }
 
+func ExtractFromMap[T any](targetMap map[string]any, fieldName string) (T, error) {
+	result, ok := targetMap[fieldName].(T)
+	if !ok {
+		err := errors.New("Failed extracting field " + fieldName + "from a given map")
```

Comment:
would there be any value including the map itself in the error message as well ?